lib/posix/options/Kconfig.pipe

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2024, Bruce Rosier
+#
+# SPDX-License-Identifier: Apache-2.0
+
+menu "POSIX Pipe Options"
+
+config POSIX_PIPE
+	bool"POSIX pipe support"
+	default y
+	depends on POSIX_API
+
+config POSIX_NUM_PIPES
+	int "Number of available pipes"
+	default 2
+	depends on POSIX_API
+	help
+	Specify the number of POSIX pipes available in the system.
+	This value controls how many pipes can be created concurrently.
```

Comment:
It would be better if this option was named somewhat more conventionally like other `*_MAX` values.

See e.g.
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/posix/sys/sysconf.h

Additionally, you will need a Kconfig option for `POSIX_PIPE_BUF`, as well as a feature test macro and sysconf enumeration.

See e.g.
https://github.com/zephyrproject-rtos/zephyr/compare/main...cfriedt:zephyr:posix-pipes2